README.md

```diff
@@ -1,2 +1,11 @@
 # comp110-worksheet-C
 Base repository for COMP110 worksheet C
+* (a) The algorithm is testing whether there are 2 or more of the same elements in a list.
+* (b) The algorithm worst case is quadratic because it checks each element of i against each element of j n times (from the same list of length n). For every i (n number of times) it goes looping through j n times. O(n * n). The j loop is nested inside the i loop.
+* (c) The algorithm still works because j is still comparing to the elements the i loop has gone through already.
+* (d) The j for loop is only checking against the elements the i loop has been through already rather than comparing every position in the list in both the i and j loops. This halves the number of iterations.
+* (e) It is still quadratic, the for j loop is still nested in the i loop but it only does half the number of iterations.
```

Comment:
✔️ though important to emphasise that the inner loop does half the number of iterations **on average** -- the actual number of iterations varies from 0 to n-1 depending on the value of i